pulp_rpm/tests/functional/api/test_copy.py

```diff
@@ -265,3 +273,138 @@ def test_all_content_recursive(self):
         dc = self.client.get(f"{RPM_CONTENT_PATH}?repository_version={latest_href}")
         dest_content = [c["pulp_href"] for c in dc["results"]]
         self.assertEqual(sorted(rpms_to_copy), sorted(dest_content))
+
+
+class StrictPackageCopyTestCase(unittest.TestCase):
+    """Test strict copy of package and its dependencies."""
+
+    @classmethod
+    def setUpClass(cls):
+        """Create class-wide variables."""
+        cls.cfg = config.get_config()
+        cls.client = gen_rpm_client()
+        cls.repo_api = RepositoriesRpmApi(cls.client)
+        cls.remote_api = RemotesRpmApi(cls.client)
+        cls.rpm_content_api = ContentPackagesApi(cls.client)
+        delete_orphans(cls.cfg)
+
+    def test_sctrict_copy_package_to_empty_repo(self):
+        """Test copy package and its dependecies to empty repository.
+
+        - Create repository and populate it
+        - Create empty repository
+        - Use 'copy' to copy 'whale' package with dependencies
+        - assert package and its dependencies were copied
+        """
+        empty_repo = self.repo_api.create(gen_repo())
+        self.addCleanup(self.repo_api.delete, empty_repo.pulp_href)
+
+        repo = self.repo_api.create(gen_repo())
+        self.addCleanup(self.repo_api.delete, repo.pulp_href)
+
+        body = gen_rpm_remote(url=RPM_UNSIGNED_FIXTURE_URL)
+        remote = self.remote_api.create(body)
+        self.addCleanup(self.remote_api.delete, remote.pulp_href)
+
+        repository_sync_data = RpmRepositorySyncURL(remote=remote.pulp_href)
+        sync_response = self.repo_api.sync(repo.pulp_href, repository_sync_data)
+        monitor_task(sync_response.task)
+
+        repo = self.repo_api.read(repo.pulp_href)
+        whale_package_href = [
+            pkg
+            for pkg in get_content(repo.to_dict())['rpm.package']
+            if pkg['name'] == 'whale'
```

Comment:
How about instead of hardcoding "whale", "stork", and "shark" into the test, we create a variable for the package name to copy ("whale") and a variable for its dependencies stored as a list or set (```{"stork", "shark"}```)

That way it's easier to change which packages are under test.

Reply:
Eventually we might even move towards completely parameterizing these tests, I think that would be a good thing to shoot for.

Reply:
I agree, I haven't put them to constans but to class variables as they are pretty much connected to this test class.